docs/pages/server-access/guides/ec2-discovery.mdx

```diff
@@ -0,0 +1,277 @@
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ssmmessages:CreateControlChannel",
+                "ssmmessages:CreateDataChannel",
+                "ssmmessages:OpenControlChannel",
+                "ssmmessages:OpenDataChannel"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ec2messages:AcknowledgeMessage",
+                "ec2messages:DeleteMessage",
+                "ec2messages:FailMessage",
+                "ec2messages:GetEndpoint",
+                "ec2messages:GetMessages",
+                "ec2messages:SendReply"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+## Step 4/5. Configure Teleport to discover EC2 instances
+
+The Discovery Service requires a valid auth token to connect to the cluster. Generate one by running the following command against your Teleport Auth Service and save it in `/tmp/token` on the node that will run the Discovery Service:
+
+```code
+tctl tokens add --type=discovery
+```
+
+In order to enable EC2 instance discovery the `discovery_service` must
+include an AWS section configured with at least one entry:
+
+```yaml
+version: v2
+teleport:
+  join_params:
+    token_name: "/tmp/token"
+    method: token
+  auth_servers:
+  - "teleport.example.com:3080"
+auth_service:
+  enabled: off
+proxy_service:
+  enabled: off
+ssh_service:
+  enabled: off
+discovery_service:
+  enabled: "yes"
+  aws:
+   - types: ["ec2"]
+     regions: ["eu-central-1"]
+     tags:
+       "env": "prod" # Match EC2 instances where tag:env=prod
+```
+
+Full documentation on EC2 discovery configuration can be found [here](../../reference/config.mdx).
+
+Once Teleport is configured for discovery, it can be started and EC2
+instances matching the tags specified in the AWS section will begin to
+be added to the Teleport cluster automatically.
+
+## Step 5/5. [Optional] Customize the default installer script
+
+To customize the default installer script, execute the following
+```code
+tctl get installer/default-installer > teleport-default-installer.yaml
+```
+The resulting `teleport-default-installer.yaml` can the be edited to
+change what gets executed when enrolling discovered EC2 instances.
+
+After making the desired changes to the default installer, the
+resource can be updated by executing:
+
+```code
+tctl create teleport-default-installer.yaml
+```
+
+Multiple installer resources can exist and be specified in the
+`install/script_name` section:
+```yaml
+discovery_service:
+  aws:
+    - types: ["ec2"] 
+      tags:
+       - "env": "prod"
+      install: # optional section when default-installer is used.
+        script_name: "default-installer" 
+    - types: ["ec2"] 
+      tags:
+       - "env": "devel"
+      install:
+        script_name: "devel-installer"		
+```
+
+---
+
+The installer resource has the following templating options:
```

Comment:
Also, I'm not sure I know what this means—can you include an example of adding these template variables to an `installer` resource?